proposals/cel_finegrained_exceptions.md

```diff
@@ -0,0 +1,320 @@
+apiVersion: policies.kyverno.io/v1alpha1
+kind: ValidatingPolicy
+metadata:
+  name: "disallow-privilege-escalation"
+spec:
+  matchConstraints:
+    resourceRules:
+    - apiGroups:   [""]
+      apiVersions: ["v1"]
+      operations:  ["CREATE", "UPDATE"]
+      resources:   ["pods"]
+  variables:
+    # list of excluded images
+    - name: excludedImages
+      expression: "['busybox:latest', 'nginx:latest']"
+  validations:
+    - expression: >
+        object.spec.containers.all(container,
+          container.image in variables.excludedImages ||
+          (
+            has(container.securityContext) &&
+            has(container.securityContext.allowPrivilegeEscalation) &&
+            container.securityContext.allowPrivilegeEscalation == false
+          )
+        )
+```
+
+
+2. **Filtering the Containers**:
+
+    This method uses a variable to create a pre-filtered list of items that the validation rule should apply to. The logic is: "Create a list of containers that are **NOT** on the exclusion list, and then validate that every container in this new list is compliant."
+
+```yaml
+apiVersion: policies.kyverno.io/v1alpha1
+kind: ValidatingPolicy
+metadata:
+  name: "disallow-privilege-escalation"
+spec:
+  matchConstraints:
+    resourceRules:
+    - apiGroups:   [""]
+      apiVersions: ["v1"]
+      operations:  ["CREATE", "UPDATE"]
+      resources:   ["pods"]
+  variables:
+    # create a new list of containers, excluding those with specified images
+    - name: containersToValidate
+      expression: >
+        object.spec.containers.filter(container, !(container.image in ["nginx:latest", "busybox:latest"]))
+  validations:
+    - expression: >
+        variables.containersToValidate.all(container,
+          has(container.securityContext) &&
+          has(container.securityContext.allowPrivilegeEscalation) &&
+          container.securityContext.allowPrivilegeEscalation == false
+        )
+```
+
+For fine-grained exceptions, we need to implement it in a similar way to the above examples. We will introduce a new field `images` in the PolicyException resource, which will allow users to specify a list of image names that should be exempt from policy enforcement.
+
+```yaml
+apiVersion: policies.kyverno.io/v1alpha1
+kind: PolicyException
+metadata:
+  name: image-based-exceptions
+spec:
+  policyRefs:
+    - name: disallow-privilege-escalation
+      kind: ValidatingPolicy
+  # list of image names to exclude
+  images:
+  - busybox:latest
+  - nginx:latest
+```
+
+In order to use the excluded images from the exception in the policy, Kyverno will introduce a new CEL object `kyverno.excludedImages` that will be available in the CEL expressions. This object will contain the list of images specified in the PolicyException resource. If no exception exists, `kyverno.excludedImages` is an empty list, and the policy functions normally.
+
+The policy can then be modified to use this new object in the validation expressions as follows
```

Comment:
Who modifies the policy?

Does Kyverno automatically do this, or does the user need to modify it? 

If Kyverno can automatically modify (preferred), will we keep the modified policy in the `status` like with autogen? We could potentially have a `status.exceptions.images` and `status.exceptions.values` or simply have the modified CEL in the status?

For other exceptions, we automatically filter out during the `match`. 

How do we automatically do this